prost-derive/src/field/mod.rs

```diff
@@ -172,6 +173,32 @@ impl Field {
             _ => None,
         }
     }
+
+    pub fn json(&self) -> Option<Option<&Json>> {
```

Comment:
Wouldn't it make more sense to flatten this into `Option<&Json>`

Reply:
Conceptionally yes, but this is actually an actual tri-state kind of thing, because (iirc) we are skipping code generation for `Self::OneOf` fields. But we could also make this more flat and just add another check for an oneof field at the callsite 🤷‍♂️.